src/shared/components/exonsCharts/ExonsChartStore.ts

```diff
@@ -0,0 +1,260 @@
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ **/
+
+import { Gene } from '../../api/generated/CBioPortalAPI';
+import { PfamDomain } from '../../api/generated/GenomeNexusAPI';
+import {
+    EnsemblTranscriptExt,
+    ExonRangeExt,
+    PfamDomainRangeExt,
+    StructuralVariantExt,
+    UtrType
+} from '../../model/Fusion';
+import MobxPromise from 'mobxpromise';
+import { remoteData } from '../../../public-lib';
+import * as _ from 'lodash';
+
+export class ExonsChartStore {
+
+    constructor(public gene: Gene,
+                public transcripts: MobxPromise<EnsemblTranscriptExt[]>,
+                public pfamDomains: MobxPromise<PfamDomain[]>,
+                public fusions: StructuralVariantExt[]
+    ) {}
+
+    readonly transcriptById = remoteData({
+        await: () => [this.transcripts],
+        invoke: () => {
+            const withExons = (this.transcripts.result || []).map(ExonsChartStore.addExonProps);
+            return Promise.resolve(_.groupBy(withExons, t => t.transcriptId))
+        }
+    });
+
+    readonly pfamDomainById = remoteData({
+        await: () => [this.pfamDomains],
+        invoke: () => Promise.resolve(_.groupBy(this.pfamDomains.result!, p => p.pfamAccession))
+    });
+
+    getExonsBySite(siteId: number, transcriptId: string, breakpoint: number): Array<ExonRangeExt> {
+        let exons: Array<ExonRangeExt> = [];
+        if (siteId === 1 || siteId === 2) {
+            const {result} = this.transcriptById;
+            if (result) {
+                const sortedExons = <ExonRangeExt[]> result[transcriptId][0].exons.sort(
+                    // sort exons by rank
+                    (exon1: ExonRangeExt, exon2: ExonRangeExt) => exon1.rank - exon2.rank
+                );
+                if (siteId === 1) {
+                    exons = sortedExons.filter(exon => exon.rank <= breakpoint);
+                } else if (siteId === 2) {
+                    exons = sortedExons.filter(exon => exon.rank >= breakpoint);
+                }
+            }
+        }
+        return exons;
+    }
+
+    static getTotalWidth(exons: ExonRangeExt[]): number {
+        return exons.reduce((accumulator, e) => {
+            e.width = e.width ? e.width : 0;
+            return accumulator + e.width;
+        }, 0);
+    }
+
+    assignExonsToFusions (fusion: StructuralVariantExt): StructuralVariantExt {
+        let site1Exons = this.getExonsBySite(1, fusion.site1EnsemblTranscriptId, fusion.site1Exon);
+        let site2Exons = this.getExonsBySite(2, fusion.site2EnsemblTranscriptId, fusion.site2Exon);
+        const exons = site1Exons.concat(site2Exons);
+        const totalWidth = ExonsChartStore.getTotalWidth(site1Exons.concat(site2Exons) || []);
+        return Object.assign({}, fusion, {
+            isLeftAligned: (fusion.site1HugoSymbol === this.gene.hugoGeneSymbol),
+            exons: exons,
+            totalWidth: totalWidth
```

Comment:
small thing: when the variable name is the same as the object name, you dont' have to repeat it.
i.e. this can be written
```
{
  isLeftAligned: ..,
  exons, 
  totalWidth
}
```